MAUI/DataGrid/Row-height-Customization.md

```diff
@@ -0,0 +1,153 @@
+
+To customize the header row height, follow the code example:
+
+{% highlight c# %}
+//Customizing header row height in SfDataGrid
+dataGrid.HeaderRowHeight = 50; 
+{% endhighlight %}
+
+## Customize RowHeight for all rows
+
+The data grid allows you to customize the height of grid rows in the scrolling region by setting the [SfDataGrid.RowHeight](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_RowHeight) property. The default value of this property is 47. This property responds to runtime changes, so it can be customized. Setting this property will change the height of all the rows in a body region with a common value. Setting `SfDataGrid.RowHeight` to zero will collapse all rows in the grid.
+
+To customize header row height, follow the code example:
+
+{% highlight c# %}
+//Customizing row height in SfDataGrid
+dataGrid.RowHeight = 60;
+{% endhighlight %}
+
+## QueryRowHeight
+
+The [SfDatagrid.QueryRowHeight](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_QueryRowHeight) event returns row heights on demand. This event receives two arguments, namely the sender handles the data grid, and the [DataGridQueryRowHeightEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridQueryRowHeightEventArgs.html). The `DataGridQueryRowHeightEventArgs` has the following properties:
+
+ * [RowIndex](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridQueryRowHeightEventArgs.html#Syncfusion_Maui_DataGrid_DataGridQueryRowHeightEventArgs_RowIndex): This property helps to identify a particular row in the grid.
+ * [Height](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridQueryRowHeightEventArgs.html#Syncfusion_Maui_DataGrid_DataGridQueryRowHeightEventArgs_Height): This property sets and returns the height of a grid row on demand. Default line size of the rows is 47.
+ * `Handled`: This property decides whether the specified height can be set to the row or not. The default value is `false`. When this property is not set, the decided height is not set to the row.
+
+To hook the `SfDatagrid.QueryRowHeight` event, and customize height of a row, follow the code example:
+
+{% highlight c# %}
+//Hooks QueryRowHeight event in SfDataGrid
+dataGrid.QueryRowHeight += DataGrid_QueryRowHeight;
+
+//Event to set the row height on demand
+private void DataGrid_QueryRowHeight(object sender, DataGridQueryRowHeightEventArgs e)
+{
+    //Sets height of the fifth row
+    if (e.RowIndex == 5)
+    {
+        e.Height = 100;
+        e.Handled = true;
+    }
+} 
+{% endhighlight %}
+
+The data grid allows you to customize the height of grid rows with various customizing options while auto calculating the row height based on the content.
+
+## Auto fit the grid rows based on content
```

Comment:
@NirmalKumarYuvaraj  - Auto fit the data rows based on content

Reply:
Changed the title.